rust/otap-dataflow/crates/channel/src/mpmc.rs

```diff
@@ -631,17 +648,18 @@ mod tests {
         });
 
         rt.block_on(local);
+        rt.block_on(handle).expect("Test task failed");
     }
 
     #[test]
     fn test_error_propagation() {
         let rt = create_test_runtime();
         let local = tokio::task::LocalSet::new();
 
-        _ = local.spawn_local(async {
+        let handle = local.spawn_local(async {
             // Test 1: SendError::Full propagation
             {
-                let (tx, _) = Channel::new(NonZeroUsize::new(1).unwrap());
+                let (tx, _rx) = Channel::new(NonZeroUsize::new(1).unwrap());
```

Comment:
This test is good candidate to explore the issue being solved in the PR. When we use `_`, Rust immediately drops the value which means in our case the receiver was being dropped as soon the channel got created. The drop implementation of Receiver would close the channel, thereby, making the next `tx.send(1)` result in a failure. The assertion that immediately follows `assert!(result.is_ok());` was failing but we never caught it because we only blocked on the local set and not on the testing task.

By changing `_` to `_rx`, the receiver is not immediately dropped, and the sender is allowed to send messages.